src/newgrf.cpp

```diff
@@ -2594,7 +2594,7 @@ static ChangeInfoResult TownHouseChangeInfo(uint hid, int numinfo, int prop, Byt
 					} else {
 						housespec->accepts_cargo[j] = cargo;
 					}
-					housespec->accepts_cargo_label[j] = CT_INVALID;
+					if (j < std::size(housespec->accepts_cargo_label)) housespec->accepts_cargo_label[j] = CT_INVALID;
```

Comment:
You didn't do this for industry var10 and var11.
And the extra check is not needed as the array size is `HOUSE_ORIGINAL_NUM_ACCEPTS`.